docs/proposals/18-shoot-CA-rotation.md

```diff
@@ -0,0 +1,195 @@
+
+### Goals
+- Offer a trigerable and safe solution to rotate all CAs in a shoot cluster.
+- Offer a process that is easily understandable for developers and stakeholders
+- Rotate the different CAs in the shoot with a similiar process to reduce complexity.
+
+### Non-Goals
+
+- Creating a process that runs fully automated without shoot owner interaction. As the shoot owner controls some secrets that would probably not even be possible.
+- Forcing the shoot owner to rotate after a certain time period. Our goal rather is to issue long running certificates and let the user decide depending on their requirements to rotate as needed.
+
+## Proposal
+
+#### kube-api server CA rotation
+
+The propsal section includes a detailed description of all steps involved for rotating from a given `CA0` to the target `CA1`.
+
+`t0`: Today's situation
+
+- `kube-apiserver` uses SERVER CERT signed by `CA0` and trusts CLIENT CERTS signed by `CA0`
+- `kube-controller-manager` issues new CLIENT CERTS with `CA0`
+- kubeconfig trusts only `CA0`
+- `ServiceAccount` secrets trust only `CA0`
+- kubelet uses CLIENT CERT signed by `CA0`
+
+`t1`: Shoot owner triggers first step of CA rotation process (--> phase one is started):
+
+- Generate `CA1`
+- Generate `CLIENT_CA1`
+- Update `kube-apiserver`, `kube-scheduler`, etc. to trust CLIENT CERTS signed by both `CA0` and `CLIENT_CA1` (`--client-ca-file` flag)
+- Update `kube-controller-manager` to issue new CLIENT CERTS now with `CLIENT_CA1`
+- Update kubeconfig so that its CA bundle contains both `CA0` and`CA1` (if kubeconfig still contains a legacy CLIENT CERT then rotate the kubeconfig)
+- Update `kube-controller-manager`to include   both `CA0` and `CA1` in `ServiceAccount` secrets. Then update `Service Account` secrets.
+- Restart control plane components so that their CA bundle contains both `CA0` and `CA1` 
+- Renew CLIENT CERTS (sign them with `CLIENT_CA1`) for control plane components (Prometheus, DWD, legacy VPN)
+- Trigger node rollout
+  - This issues new CLIENT CERTS for all kubelets signed by `CLIENT_CA1`
+  - This restarts all `Pod`s and propagates `CA0` and `CA1` into their mounted `ServiceAccount` secrets (note CAs can not be reloaded by go client, therefore we need a restart of pods.)
+- _Ask user to exchange all their client credentials (kubeconfig, CLIENT CERTS issued by `CertificateSigningRequest`s) to trust both CA0 and CA1_
+
+`t2`: Shoot owner triggers second step of CA rotation process (--> phase two is started):
+
+​	Prerequisite: The shoot owner has guaranteed that they exchanged their client credentials and triggered step 2 via an annotation.
```

Comment:
And another one (prerequisite) is of course that we have completed the operations above after `t1` was triggered, e.g. the node roll-out. Basically, but it's so obvious that I guess that's why it's not in the list, `t2` becomes possible/enabled only after that. But still, I guess it's helpful to know how this prerequisite check is implemented.